### DISCUSSION TITLE
Get-D365Module returning models in dependency order

### DISCUSSION THREAD

**Discussion by goshoom** [d365collaborative/d365fo.tools#681] [Ideas]
A useful extension of `Get-D365Module` would be returning modules in the dependency order, i.e. in the order in which they should be compiled (starting from Application Platform, which doesn't depend on any other module, continuing with models depending on Application Platform only and so on).
This won't be a huge change.  `Get-D365Module` needs a new parameter for that and if it's set, we'll call 
```PowerShell
$metadataProviderViaRuntime.ModelManifest.ListModulesInDependencyOrder()
```
instead of
```PowerShell
$metadataProviderViaRuntime.ModelManifest.ListModules()
```
Then we'll need to change the logic at the end of the function, where the list of modules gets sorted.

**Comment by Splaxi** [comment]
Awesome idea 🤘

**Reply by goshoom** [reply]
I'm going to do it. I've already made some changes, but with no testing so far.

**Reply by goshoom** [reply]
For reference, it's implement in https://github.com/d365collaborative/d365fo.tools/commit/365ff3b75618173a1c02e4968c9e7194747d3379.
